doc/sphinx/Pacemaker_Development/c.rst

```diff
@@ -371,72 +442,163 @@ Control Statements (if, else, while, for, switch)
            break;
        default:
            command3;
+           break;
    }
 
+
 .. index::
-   pair: C; operator
+   single: C; struct
 
-Operators
+Structures
+##########
+
+Changes to structures defined in public API headers (adding or removing
+members, or changing member types) are generally not possible without breaking
+API compatibility. However, there are exceptions:
+
+* Public API structures can be designed such that they can be allocated only
+  via API functions, not declared directly or allocated with standard memory
+  functions using ``sizeof``.
+
+  * This can be enforced simply by documentating the limitation, in which case
+    new ``struct`` members can be added to the end of the structure without
+    breaking compatibility.
+
+  * Alternatively, the structure definition can be kept in an internal header,
+    with only a pointer type definition kept in a public header, in which case
+    the structure definition can be changed however needed.
+
+
+.. index::
+   single: C; enum
+
+Enumerations
+############
+
+* New values should usually be added to the end of public API enumerations,
+  because the compiler will define the values to 0, 1, etc., in the order
+  given, and inserting a value in the middle would change the numerical values
+  of all later values, breaking code compiled with the old values. However, if
+  enum numerical values are explicitly specified rather than left to the
+  compiler, new values can be added anywhere.
+
+* When defining constant integer values, enum should be preferred over
+  ``#define`` or ``const`` when possible. This allows type checking without
+  consuming memory.
+
```

Comment:
This is just personal preference, but I'd like to see enums and #define blocks given with the `(1 << 5)` style instead of the `0x0000400` style.  It's just more readable to me.

Reply:
Ah yes, I did mean to add that

Reply:
Added a new commit on top